TauAnalysis/MCEmbeddingTools/python/customisers.py

```diff
@@ -0,0 +1,369 @@
+to_bemanipulate.append(module_manipulate(module_name = 'gedPhotonsTmp', manipulator_name = "Photon", steps = ["SIM", "MERGE"]))
+to_bemanipulate.append(module_manipulate(module_name = 'particleFlowTmp', manipulator_name = "PF", steps = ["SIM", "MERGE"], instance=["","CleanedHF","CleanedCosmicsMuons","CleanedTrackerAndGlobalMuons","CleanedFakeMuons","CleanedPunchThroughMuons","CleanedPunchThroughNeutralHadrons","AddedMuonsAndHadrons"]))
+
+
+to_bemanipulate.append(module_manipulate(module_name = 'ecalRecHit', manipulator_name = "EcalRecHit", instance= ["EcalRecHitsEB","EcalRecHitsEE"]))
+to_bemanipulate.append(module_manipulate(module_name = 'ecalPreshowerRecHit', manipulator_name = "EcalRecHit", instance= ["EcalRecHitsES"]))
+
+to_bemanipulate.append(module_manipulate(module_name = 'hbheprereco', manipulator_name = "HBHERecHit"))
+to_bemanipulate.append(module_manipulate(module_name = 'hbhereco', manipulator_name = "HBHERecHit"))
+to_bemanipulate.append(module_manipulate(module_name = 'zdcreco', manipulator_name = "ZDCRecHit"))
+
+to_bemanipulate.append(module_manipulate(module_name = 'horeco', manipulator_name = "HORecHit"))
+to_bemanipulate.append(module_manipulate(module_name = 'hfreco', manipulator_name = "HFRecHit"))
+to_bemanipulate.append(module_manipulate(module_name = 'castorreco', manipulator_name = "CastorRecHit"))
+
+
+to_bemanipulate.append(module_manipulate(module_name = 'dt1DRecHits', manipulator_name = "DTRecHit"))
+to_bemanipulate.append(module_manipulate(module_name = 'dt1DCosmicRecHits', manipulator_name = "DTRecHit"))
+
+to_bemanipulate.append(module_manipulate(module_name = 'csc2DRecHits', manipulator_name = "CSCRecHit"))
+to_bemanipulate.append(module_manipulate(module_name = 'rpcRecHits', manipulator_name = "RPCRecHit"))
+
+
+def modify_outputModules(process, keep_drop_list = [], module_veto_list = [] ):
+	outputModulesList = [key for key,value in process.outputModules.iteritems()]
+	for outputModule in outputModulesList:
+		if outputModule in module_veto_list:
+			continue
+		outputModule = getattr(process, outputModule)
+		for add_element in keep_drop_list:
+			outputModule.outputCommands.extend(add_element)
+	return process
+
+
+
+################################ Customizer for Selecting ###########################
+
+def keepSelected(dataTier):
+	 ret_vstring = cms.untracked.vstring(
+			 "keep *_patMuonsAfterID_*_"+dataTier,
+			 "keep *_slimmedMuons_*_"+dataTier,
+			 "keep *_selectedMuonsForEmbedding_*_"+dataTier,
+			 "keep recoVertexs_offlineSlimmedPrimaryVertices_*_"+dataTier,
+			 "keep *_firstStepPrimaryVertices_*_"+dataTier,
+			 "keep *_offlineBeamSpot_*_"+dataTier)
+	 for akt_manimod in to_bemanipulate:
+		if "CLEAN" in akt_manimod.steps:
+			ret_vstring.append("keep *_"+akt_manimod.module_name+"_*_"+dataTier)
+	 return ret_vstring
+
+def customiseSelecting(process,reselect=False):
+	if reselect:
```

Comment:
it may be more straightforward to define this as a skim instead of "customise".
Something to consider for the future.